docs/guides/12-add-widgets-advanced.md

```diff
@@ -0,0 +1,262 @@
+
+### Histogram expressions
+
+There are two histogram expressions in CARTO VL: `viewportHistogram` and `globalHistogram`. These expressions return a list of values grouped by column. The `viewportHistogram` returns the list based on the features that are visible in the viewport, while `globalHistogram` takes into account the whole dataset.
+
+#### Drawing a bar chart
+
+In this first step, we are going to draw a bar chart showing the number of trees classified by **street side**. We are going to use the `viewportHistogram` expression.
+
+```js
+  // Define the source
+  const source = new carto.source.Dataset('vancouver_trees');
+
+  // Define the visualization
+  const viz = new carto.Viz(`
+    @v_histogram: viewportHistogram($street_side_name)
+  `);
+```
+
+Char.js library needs three arrays to draw the bar chart:
+
+* `labels`: array of string values that indicate the label of each bar.
+* `data`: array of numeric values that indicate the height of each bar.
+* `backgroundColor`: array of colors that will be applied to the chart bars from left to right. If you assing a single color, all the chart bars will be colored the same.
+
+We are going to use this configuration by default:
+
+```js
+const chartOptionsDefault = {
+  animation: {
+    duration: 0
+  },
+  legend: {
+    display: false
+  },
+  scales: {
+    yAxes: [{
+      gridLines: {
+        drawBorder: false,
+        display: false
+      },
+      ticks: {
+        suggestedMin: 1,
+        beginAtZero: true,
+        display: false
+      }
+    }],
+    xAxes: [{
+      gridLines: {
+        drawBorder: false,
+        display: false
+      },
+      ticks: {
+        display: false
+      },
+      barPercentage: 1.0,
+      categoryPercentage: 1.0
+    }]
+  }
+};
+
+const chart = new Chart(ctx, {
+  type: 'bar',
+  options: chartOptionsDefault
+});
+```
+
+The `histogram.value` returns an array of `{ x, y }` objects where `x` is the name of the category and `y` is the amount of trees for this category. We are going to build `data` and `labels` arrays from the histogram expression  **once the layer is updated**:
+
+```js
+layer.on('updated', () => {
+  // Get histogram data
+  const histogramData = layer.viz.variables.v_histogram.value;
+
+  // Chart.js set up
+  const labels = histogramData.map(elem => elem.x);
+  const data = histogramData.map(elem => elem.y);
+  const backgroundColor = ['#00718b', '#089099', '#46aea0', '#7ccba2', '#f7feae'];
+
+  chart.data = {
+    labels,
+    datasets: [{
+      data,
+      backgroundColor
+    }]
+  };
+
+  chart.update();
+});
+```
+
+Here you can see the result. Check how the bars change when you interact with the map.
+
+<div class="example-map">
+  <iframe
+    id="guides-widgets-advanced-step-1"
+    src="/developers/carto-vl/examples/maps/guides/add-widgets-advanced/step-1.html"
```

Comment:
In this first example all the bars have the same teal color, so I guess that the previous code doesn't correspond to thist first step.